tests/integration/events.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"time"
+
+	"github.com/sourcenetwork/immutable"
+	"github.com/stretchr/testify/require"
+
+	"github.com/sourcenetwork/defradb/event"
+)
+
+// eventTimeout is the amount of time to wait
+// for an event before timing out
+const eventTimeout = 1 * time.Second
+
+// waitForNetworkSetupEvents waits for p2p topic completed and
+// replicator completed events to be published on the local node event bus.
+func waitForNetworkSetupEvents(s *state, nodeID int) {
+	cols, err := s.nodes[nodeID].GetAllP2PCollections(s.ctx)
+	require.NoError(s.t, err)
+
+	reps, err := s.nodes[nodeID].GetAllReplicators(s.ctx)
+	require.NoError(s.t, err)
+
+	p2pTopicEvents := 0
```

Comment:
question: It looks like `p2pTopicEvents` is only ever 0 or 1, why is it not a boolean? Is this a bug/wip?

Reply:
Changed to bool